### ISSUE TITLE
Failed to create an application gateway

### ISSUE THREAD

**Issue by bingosummer** [terraform-providers/terraform-provider-azurerm#2212]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.17.0

### Affected Resource(s)

azurerm_application_gateway.azure_application_gateway

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Actual Behavior
```
azurerm_application_gateway.azure_application_gateway: Creating...
  backend_address_pool.#:                            "" => "1"
  backend_address_pool.0.id:                         "" => "<computed>"
  backend_address_pool.0.name:                       "" => "appGatewayBackendPool"
  backend_http_settings.#:                           "" => "1"
  backend_http_settings.0.cookie_based_affinity:     "" => "Disabled"
  backend_http_settings.0.id:                        "" => "<computed>"
  backend_http_settings.0.name:                      "" => "appGWBEHttpSettings"
  backend_http_settings.0.port:                      "" => "80"
  backend_http_settings.0.probe_id:                  "" => "<computed>"
  backend_http_settings.0.protocol:                  "" => "Http"
  backend_http_settings.0.request_timeout:           "" => "1"
  frontend_ip_configuration.#:                       "" => "1"
  frontend_ip_configuration.0.id:                    "" => "<computed>"
  frontend_ip_configuration.0.name:                  "" => "appGatewayFrontendIP"
  frontend_ip_configuration.0.private_ip_address:    "" => "<computed>"
  frontend_ip_configuration.0.public_ip_address_id:  "" => "/subscriptions/4be8920b-2978-43d7-ab14-04d8549c1d05/resourceGroups/binxicpi-paint-ox-default-rg/providers/Microsoft.Network/publicIPAddresses/azure_ip_application_gateway"
  frontend_ip_configuration.0.subnet_id:             "" => "<computed>"
  frontend_port.#:                                   "" => "1"
  frontend_port.0.id:                                "" => "<computed>"
  frontend_port.0.name:                              "" => "appGWFEHttp"
  frontend_port.0.port:                              "" => "80"
  gateway_ip_configuration.#:                        "" => "1"
  gateway_ip_configuration.0.id:                     "" => "<computed>"
  gateway_ip_configuration.0.name:                   "" => "appGatewayIpConfig"
  gateway_ip_configuration.0.subnet_id:              "" => "/subscriptions/4be8920b-2978-43d7-ab14-04d8549c1d05/resourceGroups/binxicpi-paint-ox-default-rg/providers/Microsoft.Network/virtualNetworks/azure_bosh_network/subnets/azure_subnet_3"
  http_listener.#:                                   "" => "1"
  http_listener.0.frontend_ip_configuration_id:      "" => "<computed>"
  http_listener.0.frontend_ip_configuration_name:    "" => "appGatewayFrontendIP"
  http_listener.0.frontend_port_id:                  "" => "<computed>"
  http_listener.0.frontend_port_name:                "" => "appGWFEHttp"
  http_listener.0.id:                                "" => "<computed>"
  http_listener.0.name:                              "" => "appGatewayHttpListener"
  http_listener.0.protocol:                          "" => "Http"
  http_listener.0.ssl_certificate_id:                "" => "<computed>"
  location:                                          "" => "westeurope"
  name:                                              "" => "azure_application_gateway"
  request_routing_rule.#:                            "" => "1"
  request_routing_rule.0.backend_address_pool_id:    "" => "<computed>"
  request_routing_rule.0.backend_address_pool_name:  "" => "appGatewayBackendPool"
  request_routing_rule.0.backend_http_settings_id:   "" => "<computed>"
  request_routing_rule.0.backend_http_settings_name: "" => "appGWBEHttpSettings"
  request_routing_rule.0.http_listener_id:           "" => "<computed>"
  request_routing_rule.0.http_listener_name:         "" => "appGatewayHttpListener"
  request_routing_rule.0.id:                         "" => "<computed>"
  request_routing_rule.0.name:                       "" => "HTTPrule"
  request_routing_rule.0.rule_type:                  "" => "Basic"
  request_routing_rule.0.url_path_map_id:            "" => "<computed>"
  resource_group_name:                               "" => "binxicpi-paint-ox-default-rg"
  sku.#:                                             "" => "1"
  sku.682238758.capacity:                            "" => "2"
  sku.682238758.name:                                "" => "Standard_Medium"
  sku.682238758.tier:                                "" => "Standard"
  tags.%:                                            "" => "<computed>"
azurerm_application_gateway.azure_application_gateway: Still creating... (8m30s elapsed)
azurerm_application_gateway.azure_application_gateway: Still creating... (8m40s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* azurerm_application_gateway.azure_application_gateway: 1 error(s) occurred:

* azurerm_application_gateway.azure_application_gateway: unexpected EOF
```
### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by tombuildsstuff** [comment]
hi @bingosummer 

Thanks for opening this issue :)

So that we're able to investigate this further - would you be able to provide a Terraform Configuration which reproduces this issue? If that's not possible, it should be possible to more details about what's gone wrong here in the debug logs, which can be seen by setting the environment variable `TF_LOG` to a value e.g. `1` - would you be able to take a look?

Thanks!

**Comment by bingosummer** [comment]
@tombuildsstuff Thanks for your quick response! Your response is always very helpful!

I used concourse to run terraform. When I hit the issue, the concourse worker VM is `2 CPUs + 7 GB memory`. 4 terraform processes were trying to create AppGws. Some of them succeeded, some failed. After I upgraded the VM to `4 CPUs + 28 GB memory`, the issue is gone. All 4 terraform processes can succeed to create AppGws.
I guess, maybe some resource limit (network limit or disk IOPS limit) was hit when multiple terraform were trying to create AppGws.
Since this issue doesn't block me for now, I'm going to close it. Later, if I hit the issue again, I'll re-open the issue and report it with more details.
